site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx

```diff
@@ -82,6 +85,51 @@ const CreateConsequenceNetwork = (props: CreateConsequenceNetworkProps): ReactEl
                             ]}
                         />
 
+                        {!["preprod", "prod"].includes(props.stage || "development") ? (
+                            <Checkbox<NetworkConsequence>
+                                inputName="disruptionArea"
+                                displaySize="l"
+                                hideLegend={false}
+                                hint="Which area does the network cover?"
+                                display="Disruption Area"
+                                multiple
+                                checkboxDetail={
+                                    props.disruptionAreas?.map((disruptionArea) => ({
+                                        display: `${disruptionArea.name} - ${disruptionArea.administrativeAreaCode}`,
+                                        value: disruptionArea.administrativeAreaCode,
+                                        checked:
+                                            pageState.inputs?.disruptionArea?.includes(
+                                                disruptionArea.administrativeAreaCode,
+                                            ) || false,
+                                    })) || []
+                                }
+                                stateUpdater={(value, _, checked) => {
+                                    let updatedDisruptionAreas = Array.isArray(pageState.inputs?.disruptionArea)
+                                        ? [...(pageState.inputs?.disruptionArea || [])]
+                                        : pageState.inputs?.disruptionArea
+                                        ? [pageState.inputs?.disruptionArea]
+                                        : [];
```

Comment:
Would be good to avoid a nested ternary, maybe the below code is more readable:

```suggestion
 let updatedDisruptionAreas: string[] = [];

if (pageState.inputs.disruptionArea) {
    updatedDisruptionAreas = Array.isArray(pageState.inputs.disruptionArea)
        ? pageState.inputs.disruptionArea
         : [pageState.inputs.disruptionArea];
         }
```

Reply:
done